sql/ha_partition.cc

```diff
@@ -12227,6 +12227,67 @@ int ha_partition::info_push(uint info_type, void *info)
   DBUG_RETURN(error);
 }
 
+Item* ha_partition::idx_cond_push(uint keyno, Item* idx_cond)
+{
+   DBUG_ASSERT(pushed_idx_cond == nullptr);
+   DBUG_ASSERT(pushed_idx_cond_keyno == MAX_KEY);
+
+   for (uint i= bitmap_get_first_set(&m_part_info->read_partitions);
+        i < m_tot_parts;
+        i= bitmap_get_next_set(&m_part_info->read_partitions, i))
+   {
+     // We don't support pushdown to myisam because of issues with
+     // handling null selects.  According to our KB, "There is
+     // usually little reason to use it [MyISAM] except for legacy
+     // purposes."  It seems reasonable to exclude it from future
+     // developments, like index condition pushdown.
```

Comment:
This sounds very odd. Please provide more details about "issues with handling null selects"...

Reply:
Yes, I am using the PR to make a suggestion and to receive feedback on the approach.  (I'm not sure of a better way to share diffs and get feedback, please let me know if there is a better way).

There are some MyISAM tests where we get unexpected null results simply because we support index condition pushdown in this patch.  They are:
 - parts.partition_alter1_1_2_myisam 
 - parts.partition_basic_myisam 
 - parts.partition_basic_symlink_myisam
I don't think it is correct to merely re-record them and add them to the PR.  At the same time, I didn't think it was worth supporting MyISAM because it is legacy and there is little reason to use it, according to our own documentation.  It is fine to support it of course, but I would like to know what to do to address the test failures.  Here's an example failure from `parts.partition_alter1_1_2_myisam`:
```
@@ -182,7 +182,7 @@
 f_charbig = '#NULL#';
 # check null success:    1

-# check null-1 success: 	1
+# check null-1 success: 	0
 UPDATE t1 SET f_int1 = -@max_row
 WHERE f_int1 IS NULL AND f_int2 = -@max_row AND f_char1 = CAST(-@max_row AS CHAR)
 AND f_char2 = CAST(-@max_row AS CHAR) AND f_charbig = '#NULL#';
```
To me, this looks like a wrong result.